src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIPInterfaceProperties.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Sockets;
+
+namespace System.Net.NetworkInformation
+{
+    internal class LinuxIPInterfaceProperties : IPInterfaceProperties
+    {
+        private readonly LinuxNetworkInterface _linuxNetworkInterface;
+        private readonly string _dnsSuffix;
+        private readonly UnicastIPAddressInformationCollection _unicastAddresses;
+        private readonly MulticastIPAddressInformationCollection _multicastAddreses;
+        private readonly IPAddressCollection _dnsAddresses;
+        private readonly GatewayIPAddressInformationCollection _gatewayAddresses;
+        private readonly LinuxIPv4InterfaceProperties _ipv4Properties;
+        private readonly LinuxIPv6InterfaceProperties _ipv6Properties;
+
+        public LinuxIPInterfaceProperties(LinuxNetworkInterface lni)
+        {
+            _linuxNetworkInterface = lni;
+            _dnsSuffix = GetDnsSuffix();
+            _unicastAddresses = GetUnicastAddresses();
+            _multicastAddreses = GetMulticastAddresses();
+            _dnsAddresses = GetDnsAddresses();
+            _gatewayAddresses = GetGatewayAddresses();
+            _ipv4Properties = new LinuxIPv4InterfaceProperties(_linuxNetworkInterface);
+            _ipv6Properties = new LinuxIPv6InterfaceProperties(_linuxNetworkInterface);
+        }
+
+        public override bool IsDnsEnabled
+        {
+            get
+            {
+                // /etc/nsswitchc.conf may be of use here.
+                // the absence of "dns" in that file could indicate that DNS is not enabled.
+                return DnsAddresses.Count > 0;
+            }
+        }
+
+        public override string DnsSuffix { get { return _dnsSuffix; } }
+
+        public override bool IsDynamicDnsEnabled
+        {
+            get
+            {
+                throw new NotImplementedException();
```

Comment:
I'm actually planning on replacing these with "PlatformNotSupportedException" eventually, but I have left some of them (a lot of them) as "NotImplemented" until I'm able to determine that I definitely can't get the information.